_posts/2017-03-07-hddtools.md

```diff
@@ -0,0 +1,323 @@
+
+Depending on the data license, functions can provide offline and/or on-line modes. When redistribution is allowed, a copy of the dataset is cached within the package and updated twice a year. This is the fastest option and also allows offline use of functions. When re-distribution is not allowed, only on-line mode is provided.
+
+
+## Installation 
+
+The hddtools package and the examples in this blog depend on other CRAN packages. Before attempting to install hddtools, solve any missing dependencies using the commands below:
+
+
+```r
+packs <- c("survival", "raster", "devtools")
+new_packages <- packs[!(packs %in% installed.packages()[,"Package"])]
+if(length(new_packages)) install.packages(new_packages)
+```
+
+The package is available from the Get the released version from the Comprehensive R Archive Network ([CRAN](https://cran.r-project.org/index.html)):
+
+
+
+```r
+install.packages("hddtools")
+```
+
+However, the development of this package is very dynamic. Data sources become often unavailable or migrate to new systems. URLs change and, as a consequence, functions may stop working. For this reason, I always suggest to use the development version of this package, which is available from github using [devtools](https://github.com/hadley/devtools):
+
+
+```r
+devtools::install_github("ropensci/hddtools")
+```
+
+Load the hddtools package:
+
+
+```r
+library("hddtools")
+```
+
+## Data sources and Functions
+
+In my quest for hydrological data I found that there are tons of open datasets available but the problem is that __you need to know where to look!__
+
+Here is the list of data sources available within the hddtools package:
+
+- The Koppen Climate Classification map
+- The Global Runoff Data Centre
+- NASA's Tropical Rainfall Measuring Mission 
+- SEPA river level data
+- The Top-Down modelling Working Group:
+  * Data60UK
+  * MOPEX
+
+For each source, functions are available to obtain and/or filter relevant data. Their usage is described below.
+
+
+### The Koppen Climate Classification map
+
+The Koppen Climate Classification is the most widely used system for classifying the world's climates. Its categories are based on the annual and monthly averages of temperature and precipitation. This climate classification is used, in hydrological studies, to explain the continental-scale variability in
+annual runoff. The hddtools package contains a function to identify the updated Koppen-Greiger climate zone, given a bounding box. In the example below I'm getting the climate class for my beautiful home town in Italy: Pompeii.
+
+
+```r
+# Plot Pompeii on a map 
+library(raster)
+Italy <- getData("GADM", country = "IT", level = 0)
```

Comment:
global administrative areas http://www.gadm.org/ probably - linking to that would be good